sdk/eventhub/azure-eventhub/samples/async_samples/recv_async.py

```diff
@@ -18,14 +18,36 @@
 
 
 async def on_event(partition_context, event):
+    # put your code here.
+    # If the operation is i/o intensive, async will have better performance
     print("Received event from partition: {}".format(partition_context.partition_id))
-    # Do some sync or async operations. If the operation is i/o intensive, async will have better performance
-    # print(event)
+
+
+def on_partition_initialize(partition_context):
+    # put your code here
+    print("Partition: {} has been initialized".format(partition_context.partition_id))
+
+
+def on_partition_close(partition_context, reason):
+    # put your code here
+    print("Partition: {} has been closed, reason for closing: {}".format(partition_context.partition_id,
+                                                                         reason))
+
+
+def on_error(partition_context, error):
+    # put your code here
+    print("Partition: {} met an exception during receiving: {}".format(partition_context.partition_id,
+                                                                       error))
 
 
 async def receive(client):
     try:
-        await client.receive(on_event=on_event)
+        await client.receive(
```

Comment:
I suggest merge function main() and receive(client). It's easier to understand.

Reply:
updated